benchmarks/test_replaybuffer_benchmark.py

```diff
@@ -71,22 +77,28 @@ def iterate(rb):
         [TensorDictReplayBuffer, ListStorage, SamplerWithoutReplacement, 4000],
         [TensorDictReplayBuffer, LazyMemmapStorage, SamplerWithoutReplacement, 10_000],
         [TensorDictReplayBuffer, LazyTensorStorage, SamplerWithoutReplacement, 10_000],
-        [TensorDictPrioritizedReplayBuffer, ListStorage, None, 4000],
-        [TensorDictPrioritizedReplayBuffer, LazyMemmapStorage, None, 10_000],
-        [TensorDictPrioritizedReplayBuffer, LazyTensorStorage, None, 10_000],
+        [_TensorDictPrioritizedReplayBuffer, ListStorage, None, 4000],
+        [_TensorDictPrioritizedReplayBuffer, LazyMemmapStorage, None, 10_000],
+        [_TensorDictPrioritizedReplayBuffer, LazyTensorStorage, None, 10_000],
     ],
 )
-def test_sample_rb(benchmark, rb, storage, sampler, size):
+def test_rb_sample(benchmark, rb, storage, sampler, size):
     (rb,), _ = create_rb(
-        rb=TensorDictReplayBuffer,
-        storage=ListStorage,
-        sampler=RandomSampler,
+        rb=rb,
+        storage=storage,
+        sampler=sampler,
         populated=True,
         size=size,
     )()
     benchmark(sample, rb)
 
 
+def infinite_iter(obj):
```

Comment:
It seems there exists a infinite loop here and the `test_rb_iterate` could not end the 4th test. (Possibly due to the circular sampling in `SamplerWithoutReplacement`)

The old `iterate` works fine and I think you should change this back.

Reply:
Yep, I wanted to avoid the cost of entering `__iter__` but it's still better than an infinite loop!

Reply:
In this case, the following would work:

```python 
def iterate(rb_iter):
    try:
        next(rb_iter)
    except StopIteration:
        pass
```

and inside `test_rb_iterate`:

```python
benchmark(iterate, iter(rb))
```

Reply:
The problem is that I was call iter on iter, not solving the problem that the first iter was empty. I will push a fix in a sec